src/ImageSharp/Image/PixelArea{TColor}.cs

```diff
@@ -29,6 +29,11 @@ public sealed unsafe class PixelArea<TColor> : IDisposable
         private IntPtr dataPointer;
 
         /// <summary>
+        /// True if <see cref="Bytes"/> was rented from <see cref="BytesPool"/> by the constructor
+        /// </summary>
+        private bool isBufferRented;
```

Comment:
Really need to sleep, just made the field readonly for now :( Actually using a rented buffer could be the default behavior, if `Bytes` is not coming from the outside, but I wanted to keep the existing Gif & Png code safe. Confused now, we have to think this over later.